### ISSUE TITLE
ComponentField does not support nested lambda functions

### ISSUE THREAD

**Issue by jneeven** [larq/zookeeper#129]
### Describe the bug
When one has nested components, it becomes impossible to define attributes of multiple types on the innermost component. Consider the following example:
```python
    model = ComponentField(
        CustomComponent,
        some_attribute=lambda: CustomChildComponent(
            some_strings=lambda: [string_1", "string_2"],
        ),
    )
```

This will throw `TypeError: Field 'some_strings' of component 'SomeClass.model.some_attribute' is annotated with type 'typing.List[str]', which is not satisfied by value <function SomeClass.<lambda>.<locals>.<lambda> at 0x7f16c1850830>.`

Changing the outermost lambda to a `ComponentField` or `PartialComponent` raises 
```
TypeError: Keyword arguments passed to `PartialComponent` must be either:
- An immutable value (int, float, bool, string, or None).
- A function or lambda accepting no arguments and returning the 
  value that should be passed to the component upon instantiation.
- Another `PartialComponent`.Wrapping non-immutable values in a function / lambda allows the values to be lazily evaluated; they won't be created at all if the partial component is never instantiated.
```

### To Reproduce
See the code above. Sadly I don't have much time right now but I can create some example classes at a later time if necessary, they should be quite simple.

### Expected behavior
Nested lambda functions should be correctly resolved (or nested ComponentFields should be supported)

### Environment
TensorFlow version: 2.0.0
Zookeeper version: 1.0b7

**Comment by AdamHillier** [comment]
This is what `PartialComponent` is in theory designed to support without the need for lambdas, but I had mistakenly not allowed passing component classes as kwargs, or passing `PartialComponent` instances as kwargs. This will be solved by #136.

After #136 is merged, the following should work for your example above:

```python
    model = ComponentField(
        CustomComponent,
        some_attribute=PartialComponent(
            CustomChildComponent,
            some_strings=lambda: [string_1", "string_2"],
        ),
    )
```

**Comment by AdamHillier** [comment]
Closed by #136.
